src/library/searchqueryparser.cpp

```diff
@@ -128,7 +129,7 @@ void SearchQueryParser::parseTokens(QStringList tokens,
             if (!argument.isEmpty()) {
                 if (field == "crate") {
                     pNode = std::make_unique<CrateFilterNode>(
-                          m_database, argument);
+                          argument, m_pTrackCollection);
```

Comment:
The CrateFilterNode shouldn't need the TrackCollection pointer, just the CrateStorage reference.